File: leetcode/java/1480. Running Sum of 1d Array.java
class Solution {
    public int[] runningSum(int[] nums) {
        // return mySolution(nums);
        return mySolution2(nums);
    }
    
    private int[] mySolution(final int[] nums) {
        final int[] ret = new int[nums.length];
            
        for(int i = 0; i< nums.length; i++) {
            int sum = 0;
            for(int j = 0; j <= i; j++) {
                ret[i] += nums[j];
            }
        }
        
        return ret;
    }
    
    private int[] mySolution2(final int[] nums) {
        final int[] ret = new int[nums.length];
        
        ret[0] = nums[0];
        for(int i = 1; i < nums.length; i++) {
            ret[i] = ret[i - 1] + nums[i];
        }
        
        return ret;
    }
}

File: leetcode/kotlin/1480. Running Sum of 1d Array.kt
class Solution {
    fun runningSum(nums: IntArray): IntArray {
        return mySolution1(nums)
    }
    
    fun mySolution1(nums: IntArray): IntArray {
        val ret = IntArray(nums.size)
        // ret[0] = nums[0]
        ret[0] = nums.get(0)
        
        for(i in 1 until nums.size) {
            // ret[i] = ret[i-1] + nums[i]
            ret[i] = ret.get(i-1) + nums.get(i)
        }
        
        return ret
    }
}
